packages/java/engine-core/src/main/java/com/vaadin/hilla/engine/EngineConfiguration.java

```diff
@@ -2,118 +2,88 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.Arrays;
 import java.util.Collection;
-import java.util.LinkedHashSet;
-import java.util.Objects;
+import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-import com.fasterxml.jackson.annotation.JsonAutoDetect;
-import com.fasterxml.jackson.annotation.JsonIgnore;
-import com.fasterxml.jackson.annotation.PropertyAccessor;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.vaadin.flow.server.ExecutionFailedException;
 import com.vaadin.flow.server.frontend.FrontendUtils;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class EngineConfiguration {
-    public static final String DEFAULT_CONFIG_FILE_NAME = "hilla-engine-configuration.json";
+    private static EngineConfiguration INSTANCE;
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(EngineConfiguration.class);
+
     public static final String OPEN_API_PATH = "hilla-openapi.json";
-    static final ObjectMapper MAPPER = new ObjectMapper()
-            .setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE)
-            .setVisibility(PropertyAccessor.FIELD,
-                    JsonAutoDetect.Visibility.ANY);
-    private Path baseDir;
+    private Set<Path> classpath = Arrays
+            .stream(System.getProperty("java.class.path")
+                    .split(File.pathSeparator))
+            .map(Path::of).collect(Collectors.toSet());
+    private String groupId;
+    private String artifactId;
+    private String mainClass;
     private Path buildDir;
-    @JsonDeserialize(as = LinkedHashSet.class)
-    private Set<Path> classPath;
+    private Path baseDir;
     private Path classesDir;
     private GeneratorConfiguration generator;
     private Path outputDir;
     private ParserConfiguration parser;
+    private BrowserCallableFinder browserCallableFinder;
+    private boolean productionMode = false;
+    private String nodeCommand = "node";
 
     private EngineConfiguration() {
+        baseDir = Path.of(System.getProperty("user.dir"));
```

Comment:
I wonder whether this is working properly in a multi-module project, cause it depends where the JVM is started from, correct?

Reply:
This is more of control rather than a blocker: 
Even if it's not working, we can consider it as a separate issue, and fix it later, so feel free to resolve this.

Reply:
This is a default configuration which will be overridden when using the builder. Let's wait for the issue to arise.